opentracing-api/src/main/java/io/opentracing/propagation/UnsupportedFormatException.java

```diff
@@ -0,0 +1,24 @@
+/**
+ * Copyright 2016 The OpenTracing Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.opentracing.propagation;
+
+/**
+ * This exception must be thrown by Tracer's inject/extract methods
+ * when the tracer does not support provided format.
+ */
+public class UnsupportedFormatException extends RuntimeException {
```

Comment:
This should be either a checked exception, or we should be using `IllegalArgumentException`.

If it's unchecked, then it's a programmatic fault, the programmer should have implicitly or explicitly known that this format was not supported, and in that case the proper result is an `IllegalArgumentException`.

Note there is no way of explicitly checking whether the format is supported. Implicitly one knows through the configuration (registration) of the injectors and extractors. Note that calls to `extract(..)` written in libraries or frameworks are unaware of the contextual configuration to the application.

Without adding a method that can check whether a format is supported, and without making a statement that `extract(..)` called from within libraries or frameworks is not supported, I think the case is clear that this needs to be a checked exception.